src/flinkStatementResultsManager.test.ts

```diff
@@ -102,4 +102,60 @@ describe("FlinkStatementResultsManager", () => {
 
     showJsonPreviewMock.restore();
   });
+
+  it("should filter results based on search query", async () => {
+    const { manager } = await createResultsManagerWithResults();
+
+    const searchValue = "80.8";
+
+    manager.handleMessage("Search", { search: searchValue });
+
+    const filtered = manager.handleMessage("GetResults", {
+      page: 0,
+      pageSize: DEFAULT_RESULTS_LIMIT,
+    });
+
+    assert.equal(filtered.results.length, 4);
+    for (const row of filtered.results) {
+      const found = Object.values(row).some(
+        (value) =>
+          value !== null && String(value).toLowerCase().includes(searchValue.toLowerCase()),
+      );
+      assert.ok(found, `Row does not contain search value: ${JSON.stringify(row)}`);
+    }
+
+    const count = manager.handleMessage("GetResultsCount", {});
+    assert.strictEqual(count.filter, filtered.results.length);
+
+    // Clear search filter
+    manager.handleMessage("Search", { search: null });
+
+    const allResults = manager.handleMessage("GetResults", {
+      page: 0,
+      pageSize: DEFAULT_RESULTS_LIMIT,
+    });
+
+    assert.equal(allResults.results.length, 10);
+
+    const totalCount = manager.handleMessage("GetResultsCount", {});
+    assert.strictEqual(totalCount.filter, 10);
+  });
+
+  it("should filter results based on search query only in visible columns", async () => {
+    const { manager } = await createResultsManagerWithResults();
+
+    // Exists in both columns but we should only get results
+    // in the visible column `tempf`
+    const searchValue = ".8";
+
+    manager.handleMessage("Search", { search: searchValue });
+
+    const filtered = manager.handleMessage("GetResults", {
+      page: 0,
+      pageSize: DEFAULT_RESULTS_LIMIT,
+      visibleColumns: ["tempf"],
+    });
+
+    assert.equal(filtered.results.length, 4);
```

Comment:
Isn't this the same expected matching count as within test `"should filter results based on search query"` above ? Are we proving anything new other than that `handleMessage("GetResults")`  doesn't vomit if told `visibleColumns: ["tempf"]` ?

Perhaps search with different search string, then prove that it matches _only_ values within the visible column row fields?